### PR TITLE
Save comments using Datastore

### PR DESCRIPTION
**Pull request by AnudeepYakkala**
When a new comment is added, a new entity is created containing it's text and a time stamp. This entity is saved using Datastore. 

Modified the doGet() function to  query Datastore for all the stored comments and sort them by timestamp in descending order. Comments are added to an ArrayList which is returned in a json format. 

Since the comments are being stored using Datastore, they will still be there if the page is reloaded or if the server is restarted.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -44,11 +63,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
-    messages.add(comment);
+    long timestamp = System.currentTimeMillis();
 
-    // Respond with the result.
-    response.setContentType("text/html;");
-    response.getWriter().println(convertToJson(messages));
+    // Create entity with comment information
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", comment);
+    commentEntity.setProperty("timestamp", timestamp);
```
Comment (luli0521) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: nit: inline timestamp

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -25,16 +32,28 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
```
Comment (luli0521) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: Any tests for this class?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -43,12 +62,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
-    messages.add(comment);
+    String comment = getParameter(request, "text-input", "");
 
-    // Respond with the result.
-    response.setContentType("text/html;");
-    response.getWriter().println(convertToJson(messages));
+    // Create entity with comment information
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("text", comment);
+    commentEntity.setProperty("timestamp", System.currentTimeMillis());
```
[the remark this answers is not in the document]
Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I made some of the raw strings constants.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -43,12 +62,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+
+    // Store comment in Datastore
```
[the remark this answers is not in the document]
Comment (AnudeepYakkala) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Okay, I moved the comments to the JavaDoc.

**Pull request merged**